lib/autoupdate/rollout/transitions.go

```diff
@@ -0,0 +1,251 @@
+		case autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_DONE:
+			return trace.AlreadyExists("group %q is already done", groupName)
+		default:
+			return trace.BadParameter("group %q in unexpected state %s", groupName, getGroupStateName(group.GetState()))
+		}
+
+		setGroupState(group, autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_DONE, updateReasonForcedDone, now)
+	}
+
+	return nil
+}
+
+func RollbackStartedGroups(rollout *autoupdatev1pb.AutoUpdateAgentRollout, now time.Time) error {
+	groups := rollout.GetStatus().GetGroups()
+	startedGroups := make([]string, 0, len(groups))
+
+	for _, group := range groups {
+		switch group.GetState() {
+		case autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_ACTIVE,
+			autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_DONE:
+			startedGroups = append(startedGroups, group.GetName())
+		}
+	}
+
+	if len(startedGroups) == 0 {
+		return trace.NotFound("no already started group found")
+	}
+
+	return RollbackGroups(rollout, startedGroups, now)
+}
+
+// RollbackGroups mutates a rollout to rollback groups (to the rolledback state).
+// The function does not apply the rollout back, it is the caller's responsibility
+// to commit it in the backend.
+func RollbackGroups(rollout *autoupdatev1pb.AutoUpdateAgentRollout, groupNames []string, now time.Time) error {
+	// Validation part, we look for everything not in order or unsupported.
+	if rollout == nil {
+		return trace.BadParameter("rollout cannot be nil")
+	}
+
+	err := checkSchedule(rollout)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = strategySupportsManualTransition(rollout)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	groups := rollout.GetStatus().GetGroups()
+	if len(groups) == 0 {
+		return trace.BadParameter("rollout has no groups")
+	}
+
+	// Part where we do the real work, doing a state transition for every requested group.
+	for _, groupName := range groupNames {
+		group, err := getGroup(groups, groupName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		if group.GetState() == autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_ROLLEDBACK {
+			return trace.AlreadyExists("group %q is already rolledback", groupName)
+		}
+
+		setGroupState(group, autoupdatev1pb.AutoUpdateAgentGroupState_AUTO_UPDATE_AGENT_GROUP_STATE_ROLLEDBACK, updateReasonManualRollback, now)
+	}
+
+	return nil
+}
+
+// getGroup returns the group of a given name in a rollout.
+func getGroup(groups []*autoupdatev1pb.AutoUpdateAgentRolloutStatusGroup, groupName string) (*autoupdatev1pb.AutoUpdateAgentRolloutStatusGroup, error) {
+	if groupName == "" {
+		return nil, trace.BadParameter("group name cannot be empty")
+	}
+	for _, group := range groups {
+		if group.GetName() == groupName {
+			return group, nil
+		}
+	}
+	return nil, trace.NotFound("group %v not found", groupName)
```

Comment:
```suggestion
	return nil, trace.NotFound("group %q not found", groupName)
```
If you wanted to match the others, although the quotes don't look great in trace formatting